app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/TestUtils.java

```diff
@@ -59,6 +73,56 @@ public static void allowPermissionsIfNeeded() {
     }
   }
 
+  public static void captureAndSaveScreenshot(String name){
+    storeImage(
+        getOutputMediaFile(name),
+        Screenshot.capture().getBitmap()
+    );
+  }
+
+  // https://stackoverflow.com/questions/15662258/how-to-save-a-bitmap-on-internal-storage#15662384
+  private static void storeImage(File pictureFile, Bitmap image) {
+    if (pictureFile == null) {
+      Log.d(TAG,
+              "Error creating media file, check storage permissions: ");// e.getMessage());
+      return;
+    }
+    try {
+      FileOutputStream fos = new FileOutputStream(pictureFile);
+      image.compress(Bitmap.CompressFormat.PNG, 90, fos);
+      fos.close();
+    } catch (FileNotFoundException e) {
+      Log.d(TAG, "File not found: " + e.getMessage());
+    } catch (IOException e) {
+      Log.d(TAG, "Error accessing file: " + e.getMessage());
+    }
+  }
+
+  private static File getOutputMediaFile(String name){
+    // To be safe, you should check that the SDCard is mounted
```

Comment:
These comments came directly from the example code you used. They don't really fit here. If practical I suggest you move these methods to a separate file so we keep the codebase tidier (since this is enough code to be a significant part of this file and therefore shouldn't be treated as being part of the Kiwix codebase). Separating the code out also makes the overall codebase a bit cleaner and enables us to consider replacements for taking screenshots in future.